### DISCUSSION TITLE
How to gain access to written bytes to calculate checksum?

### DISCUSSION THREAD

**Discussion by yanshay** [sharksforarms/deku#499] [Q&A]
I have a struct that ends with a checksum of previous data.
I want to implement a custom writer for that field.
How can I gain in the writer access to the bytes that were already written?

In the past I did it and used `deku::output` but it seems the writer syntax has changed and now there is a `deku::writer` which doesn't seem to provide access to written bytes.

So how can I achieve something like that? It's pretty common in many binary protocols I've seen.

**Comment by wcampbell0x2a** [comment]
Take a glance at this example: https://github.com/sharksforarms/deku/blob/master/examples/deku_input.rs

**Reply by yanshay** [reply]
This example seems to be an example of reading. 
Are you saying that it's also possible to do something similar when writing? I don't fully understand that example yet, and will look closer, just want to make sure I understand correctly what you mean.

**Reply by wcampbell0x2a** [reply]
Oh, yes. Maybe I shouldn't reply right after I get up 🤷‍♂️ 

I'll have to think about the `writer`.
